enterprise/api_client/enterprise_catalog.py

```diff
@@ -169,6 +170,35 @@ def get_content_metadata(self, enterprise_customer, enterprise_catalogs=None):
 
         return list(content_metadata.values())
 
+    @JwtLmsApiClient.refresh_token
+    def refresh_catalogs(self, enterprise_catalogs):
```

Comment:
I don't think we should allow refreshing multiple catalogs in this way since it could unintentionally cause weird behavior if the catalogs share a catalog query. It's not the best UX, but I think it's reasonable to say that if you only want to update a few catalogs you just individually update them.
If we do want to support this however, I think we'd need to update the backend to take multiple catalogs and update the set of distinct catalog queries between them all

Reply:
Well is there a way to get django actions to work so that what gets acted on isn't through a bulk selection? And the catalogs sharing a query isn't an issue for the daily jobs that run?

Reply:
Sure, just give an error message if `queryset.count() > 1`. And the daily job does the idea of distinct catalogs mentioned above: https://github.com/edx/enterprise-catalog/blob/4fe13a5b20912422b5463edc720bb78af32e9862/enterprise_catalog/apps/catalog/management/commands/update_content_metadata.py#L56

Reply:
Actually how does refreshing the multiple catalogs individually by selecting them one by one address the shared catalogs weirdness issue? If I were to individually click on catalog A and refresh it and then click on catalog B and refresh it one after the other would the window of time it would take me to do the clicking be large enough that we won't have to worry about the weirdness happening like we do for making multiple api calls in a loop?

Reply:
It would still happen. I don't like either way so I don't think it _really_ matters which way we go. It just feels weirder to me to allow multiple at once when we don't really support that, especially when this seems like it would only ever be used for a single catalog at a time by ECS/developers

Reply:
I think it would be weirder to have a UI that is designed to be used for bulk selecting but we would need to somehow communicate with people to only select one catalog at a time. But even if we do communicate that we don't really have a way to say that what they should actually be doing is waiting for the previously refreshed catalog task to finish refreshing before selecting another. Because telling them to refresh them individually wouldn't actually solve the issue about worrying out data weirdness happening.

Maybe we should just change the refresh endpoint to allow for multiple catalogs so we can do what the daily job does and only refresh the distinct queries? That way we can try to avoid any data weirdness and also not have to try communicating on the front end to select things individually and then wait for the task to finish even though they might not actually know where to look to know if the task had finished.

Reply:
My two cents:
* It's an internal tool, and my _guess_ is that it will mostly be used to update only a single catalog at a time.  We can communicate to the users of this tool that they're probably better off only updating one at a time.
* I think it's reasonable for this client (or any other) to make a series of POSTs to `refresh_metadata` for a series of catalogs.
* We should modify the enterprise-catalog service to more gracefully deal with this type of situation, somehow ensuring that we don't bother updating the same piece of content metadata multiple times within a short time window.